### ISSUE TITLE
Knowledge Bases in Local

### ISSUE THREAD

**Issue by Khampol** [griptape-ai/ComfyUI-Griptape#198]
Hi,
 Is there a way to do it and host locally for griptape ? 
Sorry but i can't just host (and share) my stuff with you ;>

**Comment by griptape-studio** [comment]
Hi @Khampol - We don't have a self-install version of Griptape Cloud available for running locally, but you can use the RAG Engine node along with  the Griptape RAG Retrieve: Vector Store Module to query against your own local knowledge base. 

You would just need to have a local vector store running - here's a list of all the vector store drivers available:

![image](https://github.com/user-attachments/assets/790bba8b-1dba-4cc8-927b-347284ecc2f9)

For example, you could try MongoDB Atlas running local https://www.mongodb.com/docs/atlas/cli/current/atlas-cli-deploy-local/

Or, you could use the Local VectorStoreDriver which can use a text file - but it's not super efficient. Check out the [example](https://github.com/griptape-ai/ComfyUI-Griptape/tree/main/examples#rag-retrieval-augmented-generation) in the examples folder.

**Comment by Khampol** [comment]
Ok i see. Vector DB stuff is very new for me...
BTW do you think i could use this "[MongoDB](https://www.turnkeylinux.org/mongodb)" for it ?

**Comment by shhlife** [comment]
Yep, we have a lot of different VectorStore Drivers available - in the Griptape Docs you can see all of them:

https://docs.griptape.ai/stable/griptape-framework/drivers/vector-store-drivers/#vector-store-drivers

Check out a few of them & see which ones you feel comfortable running locally on your machine. When I'm back from the holidays I'll see if I can create a demo of one!

**Comment by Khampol** [comment]
Great! Really want to see that. 
But having the mongoDB, now having functions like https://cloud.griptape.ai/ as Create Data source or Link website etc.. How i can do that?

**Comment by shhlife** [comment]
Ah, at the moment I don't have any nodes inside ComfyUI that will create collections.. just adding data to existing collections. Creating would be things that you'd do from mongodb itself. it's something we could investigate though!

**Comment by Khampol** [comment]
Yeah, having a VectorDB GUI manager would be so great ! Cheers!

**Comment by shhlife** [comment]
Not currently planning on building a vectorDB gui manager for all the different vector dbs out there.. would just be beyond the scope of what I can handle with my time & resources. :)
